client/pages/page-edit.jsx

```diff
@@ -61,19 +40,52 @@ var PageField = React.createClass({
   propTypes: {
     page: React.PropTypes.object,
     project: React.PropTypes.object,
-    field: React.PropTypes.string,
-    update: React.PropTypes.func,
-    save: React.PropTypes.func
+    field: React.PropTypes.string
   },
   getInitialState () {
     let page = this.props.page
     let field = this.props.field
     let type = (page.schema[field] && page.schema[field].type) || 'text'
     let content = page.content.json[field]
-    return { type, content }
+    let newContent = (content instanceof Object) ? Object.assign({}, content) : content
+    return { type, content, newContent }
```

Comment:
What's the reason for differentiating between `content` and `newContent`? `content` is used in `render` but nowhere else?